crypto/threads_win.c

```diff
@@ -23,7 +23,7 @@
  * only VC++ 2008 or earlier x86 compilers.
  */
 
-#if (defined(_MSC_VER) && defined(_M_IX86) && _MSC_VER <= 1600)
+#if ((defined(_MSC_VER) && defined(_M_IX86) && _MSC_VER <= 1600) || (defined(__MINGW32__) && !defined(__MINGW64__)))
```

Comment:
Aren't __MINGW32__ and __MINGW64__ mutually exclusive?

Reply:
Also, assuming you get the ICLA signed and (potentially) fix the issue above, we can likely waive the style issues

Reply:
Running `x86_64-w64-mingw32-gcc -dM -E - < /dev/null  | grep -i GW`, I get 
```
#define __MINGW32__ 1
#define __MINGW64__ 1
```
so it doesn't seem so.

Reply:
To clarify, `x86_64-w64-mingw32-gcc` is part of MinGW-w64, not MinGW